src/node/internal/internal_fs_callback.ts

```diff
@@ -0,0 +1,1373 @@
+    // them along to the reportError function, which logs them and triggers the
+    // global "error" event.
+    // eslint-disable-next-line @typescript-eslint/no-confusing-void-expression
+    queueMicrotask(() => callback(null));
+  } catch (err) {
+    // eslint-disable-next-line @typescript-eslint/no-confusing-void-expression
+    queueMicrotask(() => callback(err));
+  }
+}
+
+function callWithSingleArgCallback<T>(
+  fn: () => T,
+  callback: undefined | SingleArgCallback<T>
+): void {
+  if (typeof callback !== 'function') {
+    throw new ERR_INVALID_ARG_TYPE('callback', ['function'], callback);
+  }
+  try {
+    const result = fn();
+    // eslint-disable-next-line @typescript-eslint/no-confusing-void-expression
+    queueMicrotask(() => callback(null, result));
+  } catch (err) {
+    // eslint-disable-next-line @typescript-eslint/no-confusing-void-expression
+    queueMicrotask(() => callback(err));
+  }
+}
+
+export function access(path: FilePath, callback: ErrorOnlyCallback): void;
+export function access(
+  path: FilePath,
+  mode: number,
+  callback: ErrorOnlyCallback
+): void;
+export function access(
+  path: FilePath,
+  modeOrCallback: number | ErrorOnlyCallback = F_OK,
+  callback?: ErrorOnlyCallback
+): void {
+  let mode: number;
+  if (typeof modeOrCallback === 'function') {
+    callback = modeOrCallback;
+    mode = F_OK;
+  } else {
+    mode = modeOrCallback;
+  }
+  // eslint-disable-next-line @typescript-eslint/no-confusing-void-expression
+  callWithErrorOnlyCallback(() => fssync.accessSync(path, mode), callback);
+}
+
+export type ExistsCallback = (result: boolean) => void;
+
+export function exists(path: FilePath, callback: ExistsCallback): void {
+  // eslint-disable-next-line @typescript-eslint/no-confusing-void-expression
+  queueMicrotask(() => callback(fssync.existsSync(path)));
+}
+
+export function appendFile(
+  path: number | FilePath,
+  data: string | ArrayBufferView,
+  callback: ErrorOnlyCallback
+): void;
+export function appendFile(
+  path: number | FilePath,
+  data: string | ArrayBufferView,
+  options: WriteFileOptions,
+  callback: ErrorOnlyCallback
+): void;
+export function appendFile(
+  path: number | FilePath,
+  data: string | ArrayBufferView,
+  optionsOrCallback: WriteFileOptions | ErrorOnlyCallback,
+  callback?: ErrorOnlyCallback
+): void {
+  let options: WriteFileOptions;
+  if (typeof optionsOrCallback === 'function') {
+    callback = optionsOrCallback;
+    options = {
+      encoding: 'utf8',
+      mode: 0o666,
+      flag: 'a',
+      flush: false,
+    };
+  } else {
+    options = optionsOrCallback;
+  }
+  callWithSingleArgCallback<number>(
+    () => fssync.appendFileSync(path, data, options),
+    callback
+  );
+}
+
+export function chmod(
+  path: FilePath,
+  mode: number,
+  callback: ErrorOnlyCallback
+): void {
+  // eslint-disable-next-line @typescript-eslint/no-confusing-void-expression
```

Comment:
Is there no validation for input, in all these functions? What about adding validate number to mode?

Reply:
The `fssync.chmodSync` will apply input validation. If we also performed input validation here we'd be duplicating.